ln-gateway/src/lib.rs

```diff
@@ -104,12 +109,15 @@ impl LnGateway {
         self.federation_client
             .claim_outgoing_contract(contract_id, preimage, rng)
             .await?;
+
+        Ok(())
+    }
+
+    pub async fn await_contract_accepted(&self, contract_id: ContractId) {
```

Comment:
That naming is confusing. The contract funding is long accepted at that point (what I used synonymous with the contract being accepted) but we want to wait for our claim transaction to be confirmed which essentially destroys the contract again. Maybe `await_contract_claimed`?